core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -265,7 +267,7 @@ private boolean isWriteSetOverlappedWith(Scan scan) {
 
   @VisibleForTesting
   void toSerializableWithExtraWrite(MutationComposer composer)
-      throws ExecutionException, CommitConflictException {
+      throws ExecutionException, PreparationConflictException {
```

Comment:
Just a confirmation. I think this is a part of pre-commit validation phase not prepare phase. Is it okay to throw PreparationConflictException?

Reply:
Yes, Extra-write is a part of validation logically, but it actually executes in prepare phase. That's why this method throws `PreparationConflictException`.